### PR TITLE
Create 1.2.2.5_Recurrence_Relation_Master_Method.md

### LINKED ISSUE #486: DSA 1.2.2.5 Algorithms: Recurrence Relations
## Description
How are recurrences solved using Master Method? Discuss the approach, applications and the efficiency.

## Task
- [ ] Documentation
- [ ] Audio
- [ ] Video

## Note
- While requesting the issue to be assigned, please mention your **NAME & BATCH NUMBER**.
- Kindly refer to the [Issue Guidelines](https://github.com/girlscript/winter-of-contributing/blob/DSA/DSA/CONTRIBUTING.md#issues).
- In case you see that the issue has already been assigned, try to request for a different task within the same issue.
- Changes should be made inside the ````DSA/directory```` & ````DSA```` branch.
- Name of the file should be same as the issue.
- This issue is only for 'GWOC' contributors of 'DSA' Domain.

### PR DESCRIPTION
**Pull request by girija0707**
<hr>

## Description 📜

Added information about recurrences solved using Master Method via documentation
Fixes #486

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] DSA

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

### CODE REVIEW

**Comment by nimisha-yadav** [comment]
Please review @harshsri28

File: DSA/1.2 Analysis of Algorithms/2. Recurrence Relations/1.2.2.5_Recurrence_Relation_Master_Method.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+T(n) = (N⁰*log(N)) = O(logN)
+
+**Example 2**
+```
+T(N) = 2*T(N/2) + O(1)
+```
+The above recurrence relation is of finding min-max in an array through the divide and conquer approach
+Comparing this with master theorem,a = 2, b = 2 and f(N) = 1,comparing left and right sides of f(N), we get k = 0.
+logb(a) = log2(2) = 1 > k so, we can apply the case 1 of the master thorem
```
Comment (harshsri28) [inline DSA/1.2 Analysis of Algorithms/2. Recurrence Relations/1.2.2.5_Recurrence_Relation_Master_Method.md:37]: highlight these text

eg-> 
`logb(a) = log2(2) = 1 > k`

File: DSA/1.2 Analysis of Algorithms/2. Recurrence Relations/1.2.2.5_Recurrence_Relation_Master_Method.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+T(N) = O(N^logb(a)) = O(N¹) = O(N)
+
+**Example 3**
+```
+T(N) = 2T(N/2) + CN^2
+```
+Comparing this with master theorem, a = 2, b = 2 and f(N) = CN²
+comparing left and right sides of f(n), we get k = 2.
+logb(a) = log2(2) = 1 < k
```
Comment (harshsri28) [inline DSA/1.2 Analysis of Algorithms/2. Recurrence Relations/1.2.2.5_Recurrence_Relation_Master_Method.md:47]: highlight these text

example->
`logb(a) = log2(2) = 1 < k`

File: DSA/1.2 Analysis of Algorithms/2. Recurrence Relations/1.2.2.5_Recurrence_Relation_Master_Method.md (hunk 1 above)
Comment (harshsri28) [inline DSA/1.2 Analysis of Algorithms/2. Recurrence Relations/1.2.2.5_Recurrence_Relation_Master_Method.md:37]: highlight these text

example->
logb(a) = log2(2) = 1 < k

**Review by harshsri28 — COMMENTED** [review]
Highlight the text

File: DSA/1.2 Analysis of Algorithms/2. Recurrence Relations/1.2.2.5_Recurrence_Relation_Master_Method.md (hunk 1 above)
Comment (girija0707) [reply DSA/1.2 Analysis of Algorithms/2. Recurrence Relations/1.2.2.5_Recurrence_Relation_Master_Method.md]: please review the changes

**Review by harshsri28 — APPROVED** [review]
LGTM

**Review by nimisha-yadav — APPROVED** [review]
LGTM. Congrats on getting your PR merged!

**Comment by girija0707** [comment]
> LGTM. Congrats on getting your PR merged!

Thank you so much ☺️

**Pull request merged**